### ISSUE TITLE
AWX inventory update blocked by itself

### ISSUE THREAD

**Issue by hungpr0** [ansible/awx#14324]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

Hello,

I'm running AWX `22.6.0`, upgraded from `22.5.0` few days ago and meet this bug. If I am lucky enough, I will see pending Inventory update jobs blocked by themselves (Ex: inventory update job has ID 137966 has pending status. When checking `Explanation`, I see: `waiting for inventoryupdate 137966 to finish`). This results related ansible jobs are blocked. New inventory update jobs might be able to run.

check out the images for detail:
https://github.com/hungpr0/img/blob/main/Screenshot_618.png
https://github.com/hungpr0/img/blob/main/Screenshot_620.png

My setup info:
- K8s cluster was deployed by k3d (k3s)
- I have job template with "Concurrent Jobs", "Provisioning Callbacks" enabled.
- The related project (git project) has "Update Revision on Luanch", "Clean" enabled.

Not sure if the issue was because of db migration or something. If anyone knows the reasons make this issue, would be awesome.

Thank you

### AWX version

22.6.0

### Select the relevant components

- [X] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

kubernetes

### Modifications

no

### Steps to reproduce

Couldn't find the way to replicate the issue.

### Expected results

inventoryupdate job shouldn't blocked by itself

### Actual results

inventoryupdate is blocked by itself randomly

**Comment by fosterseth** [comment]
so do the inventory syncs stay in pending or do they eventually run?

can you provide the `api/v2/inventory_updates/<inventory sync #>` output when this happens?

**Comment by hungpr0** [comment]
> so do the inventory syncs stay in pending or do they eventually run?
> 
> can you provide the `api/v2/inventory_updates/<inventory sync #>` output when this happens?

Hello @fosterseth ,

Thanks for your response. Here is the api info:
```
{
    "id": 1390970,
    "type": "inventory_update",
    "url": "/api/v2/inventory_updates/1390970/",
    "related": {
        "unified_job_template": "/api/v2/inventory_sources/9/",
        "stdout": "/api/v2/inventory_updates/1390970/stdout/",
        "inventory_source": "/api/v2/inventory_sources/9/",
        "cancel": "/api/v2/inventory_updates/1390970/cancel/",
        "notifications": "/api/v2/inventory_updates/1390970/notifications/",
        "events": "/api/v2/inventory_updates/1390970/events/",
        "inventory": "/api/v2/inventories/2/",
        "credentials": "/api/v2/inventory_updates/1390970/credentials/"
    },
    "summary_fields": {
        "organization": {
            "id": 2,
            "name": "My_org",
            "description": "My_org"
        },
        "inventory": {
            "id": 2,
            "name": "Production",
            "description": "Production inventory",
            "has_active_failures": true,
            "total_hosts": 194,
            "hosts_with_active_failures": 4,
            "total_groups": 108,
            "has_inventory_sources": true,
            "total_inventory_sources": 1,
            "inventory_sources_with_failures": 0,
            "organization_id": 2,
            "kind": ""
        },
        "unified_job_template": {
            "id": 9,
            "name": "Production",
            "description": "Production inventory source",
            "unified_job_type": "inventory_update"
        },
        "inventory_source": {
            "id": 9,
            "name": "Production",
            "source": "scm",
            "last_updated": "2023-08-10T02:55:31.525077Z",
            "status": "pending"
        },
        "user_capabilities": {
            "delete": true,
            "start": true
        },
        "credentials": []
    },
    "created": "2023-08-10T02:19:02.360799Z",
    "modified": "2023-08-10T02:19:02.360832Z",
    "name": "Production - Production",
    "description": "Production inventory source",
    "source": "scm",
    "source_path": "inventories/production/hosts.yml",
    "source_vars": "---",
    "scm_branch": "",
    "credential": null,
    "enabled_var": "",
    "enabled_value": "",
    "host_filter": "",
    "overwrite": true,
    "overwrite_vars": true,
    "custom_virtualenv": null,
    "timeout": 0,
    "verbosity": 1,
    "limit": "",
    "unified_job_template": 9,
    "launch_type": "callback",
    "status": "pending",
    "execution_environment": null,
    "failed": false,
    "started": null,
    "finished": null,
    "canceled_on": null,
    "elapsed": 0.0,
    "job_args": "",
    "job_cwd": "",
    "job_env": {},
    "job_explanation": "waiting for inventoryupdate-1390970 to finish",
    "execution_node": "",
    "controller_node": "",
    "result_traceback": "",
    "event_processing_finished": false,
    "launched_by": {},
    "work_unit_id": null,
    "inventory": 2,
    "inventory_source": 9,
    "license_error": false,
    "org_host_limit_error": false,
    "source_project_update": null,
    "instance_group": null,
    "scm_revision": "",
    "source_project": null
}
```
Here is output of `/api/v2/inventory_updates/1390970/events/`
```
{
    "count": 0,
    "next": null,
    "previous": null,
    "results": []
}
```

` /api/v2/inventory_updates/1390970/stdout/` is empty

The pending inventory update job `1390970` just stuck there as screenshot https://github.com/hungpr0/img/blob/main/Screenshot_321.png. The pending inventory update job is able to cancel.
Later inventory update jobs can be successful `1391091`, `1391084`...

**Comment by hungpr0** [comment]
Update info: `invetoryupdate-1396136` which had pending status forever, we have logs:

```
2023-08-11 10:32:02,685 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle job-1396135 acknowledged
2023-08-11 10:32:02,714 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle projectupdate-1396139 created
2023-08-11 10:32:02,784 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle inventoryupdate-1396136 acknowledged
2023-08-11 10:32:02,793 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle job-1396137 acknowledged
2023-08-11 10:32:02,816 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle inventoryupdate-1396136 acknowledged
2023-08-11 10:32:02,824 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle inventoryupdate-1396136 acknowledged
2023-08-11 10:32:02,967 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle job-1396135 blocked by inventoryupdate-1396136
2023-08-11 10:32:02,967 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle inventoryupdate-1396136 blocked by inventoryupdate-1396136
2023-08-11 10:32:02,968 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle job-1396137 blocked by inventoryupdate-1396136
2023-08-11 10:32:02,968 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle projectupdate-1396139 controller node chosen
2023-08-11 10:32:02,968 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle projectupdate-1396139 execution node chosen
2023-08-11 10:32:03,058 INFO     [74e9898b188e4a62898b5a9393e49efe] awx.analytics.job_lifecycle projectupdate-1396139 waiting

```

There are multiple `inventoryupdate-1396136 acknowledged` logs. All pending inventoryupdate job has these multiple log entries, while the successful job only has 1 `acknowledged` log entry. Do you know the cause of this?

**Comment by hungpr0** [comment]
From what I can check, the pending `inventoryupdate` job never creates new pod.
not sure it is related to last changes on task_manager.py or not... Anyone has an idea?

**Comment by hungpr0** [comment]
Issue seems be resolved by upgrading from `22.6.0` to `22.7.0`

**Comment by hungpr0** [comment]
No, it wasn't fixed after upgrading to `22.7.0`. I still have pending inventory update job....

**Comment by Julien5858** [comment]
We are currently getting the exact same issue running on 24.5.0 while updating an Active directory inventory
@hungpr0 did you get this issue solved somehow?

Thanks
